src/CsvHelper/CsvParser.cs

```diff
@@ -160,6 +160,16 @@ public string[] Record
 		/// <param name="configuration">The configuration.</param>
 		public CsvParser(TextReader reader, CsvConfiguration configuration)
 		{
+			if (reader is null)
+			{
+				throw new ArgumentNullException(nameof(reader));
+			}
+
+			if (configuration is null)
+			{
+				throw new ArgumentNullException(nameof(configuration));
+			}
+
 			configuration.Validate();
 
 			this.reader = reader;

```

Comment:
You can simplify the null-checks and make code more idiomatic. Try this:
```suggestion
			this.reader = reader ?? throw new ArgumentNullException(nameof(reader));
```

Reply:
See https://github.com/JoshClose/CsvHelper/pull/1822/files#r662332485

@JoshClose do you have any preference? If the shorter approach is preferred, I could take the opportunity to include a ReSharper settings file at the solution level to use this style by default moving forward.